frontstage/templates/register/register-share-survey-enter-your-details.html

```diff
@@ -0,0 +1,182 @@
+        {% endcall %}
+        <h2 class="u-mt-l">Create a password</h2>
+        <p>Your password must have:</p>
+        <ul>
+            <li>at least 8 characters </li>
+            <li>at least one symbol (eg ?!£%) </li>
+            <li>at least one capital letter</li>
+            <li>at least one number</li>
+        </ul>
+        {%- call onsFieldset({
+            "legend": "Create a password",
+            "legendClasses": "u-vh"
+        }) -%}
+            {% if errors.password %}
+                {% set errorPassword = { "text": errors['password'][0],  "id": 'password_error' } %}
+            {% endif %}
+            {{
+                onsPassword({
+                    "id": "password",
+                    "name": "password",
+                    "type": "text",
+                    "label": {
+                        "text": "Password"
+                    },
+                    "showPasswordText": "Show password",
+                    "error": errorPassword
+                })
+            }}
+            {{
+                onsPassword({
+                    "id": "password_confirm",
+                    "name": "password_confirm",
+                    "type": "text",
+                    "label": {
+                        "text": "Re-type your password"
+                    },
+                    "showPasswordText": "Show password",
+                    "error": errorPassword
+                })
+            }}
+
+        {% endcall %}
+        
+        <h2 class="u-mt-l">Contact telephone number</h2>
+        <p>We'll use this number if we need to discuss any survey information provided</p>
+        {%- call onsFieldset({
+            "legend": "Create a password",
+            "legendClasses": "u-vh"
+        }) -%}
+            {% if errors.phone_number %}
+                {% set errorPhoneNumber = { "text": errors['phone_number'][0],  "id": 'phone_number_error' } %}
+            {% endif %}
+            {{
+                onsInput({
+                    "id": "phone_number",
+                    "name": "phone_number",
+                    "type": "tel",
+                    "autocomplete": "tel",
+                    "classes": "input--w-8",
+                    "label": {
+                        "text": "Telephone number",
+                        "description": "For international numbers include the country code, for example +33 1234 567 890"
+                    },
+                    "error": errorPhoneNumber
+                })
+            }}        
+        {% endcall %}
+        <h3 class="u-mt-l">Privacy notice</h3>
+            <p>Your user account will be deleted after 36 months of inactivity</p>
+        {{
+            onsButton({
+                "text": "Continue",
+                "type": "submit",
+                "classes": "u-mt-l u-mb-s",
+                "id": "continue_button",
+                "submitType": "timer"
+            })
+        }}        
+
+    </form>
+
+    <script>
```

Comment:
Does this need csp_nonce?

Reply:
Yes. it would 👍  missed it. Thanks for it.